contrib/plugins/todo-list-backend/src/service/todos.ts

```diff
@@ -23,10 +23,41 @@ export type Todo = {
   timestamp: number;
 };
 
-export type TodoFilter = Parameters<Array<Todo>['filter']>[0];
+export type TodoFilter = {
+  property: Exclude<keyof Todo, 'timestamp'>;
+  values: Array<string | undefined>;
+};
+
+export type TodoFilters =
```

Comment:
I like that this encourages plugin authors to define their own criteria object internally rather than relying directly on `PermissionCriteria`